python/lsst/ts/MTMount/mtmount_csc.py

```diff
@@ -670,27 +651,86 @@ def terminate_background_processes(self):
+                self._camera_cable_wrap_follow_loop()
+            )
+            self.evt_cameraCableWrapFollowing.set_put(enabled=True, force_output=True)
+        except asyncio.CancelledError:
+            self.log.info("Camera cable wrap following canceled before it starts")
+            self.evt_cameraCableWrapFollowing.set_put(enabled=False)
+            raise
+        except salobj.ExpectedError as e:
+            self.log.error(f"Camera cable wrap tracking could not be enabled: {e}")
+            self.evt_cameraCableWrapFollowing.set_put(enabled=False)
+            raise
+        except Exception:
+            self.log.exception("Camera cable wrap tracking could not be enabled")
+            self.evt_cameraCableWrapFollowing.set_put(enabled=False)
+            raise
+
+    async def _camera_cable_wrap_follow_loop(self):
+        """Implement the camera cable wrap following the camera rotator.
+
+        This should be called by camera_cable_wrap_start_following.
+        Camera cable wrap tracking must be enabled before this is called.
+        """
+        self.log.info("Camera cable wrap following begins")
+        self.rotator_position_error_excessive = False
+        paused = False
         try:
             while True:
-                position_velocity_tai = await self.get_camera_cable_wrap_demand()
-                if position_velocity_tai is None:
-                    return
+                try:
+                    position_velocity_tai = await self.get_camera_cable_wrap_demand()
+                except asyncio.TimeoutError:
+                    if not paused:
+                        paused = True
+                        self.log.warning(
+                            "Rotator data not available; stopping the camera "
+                            "cable wrap until rotator data is available"
+                        )
+                        await self.send_command(commands.CameraCableWrapStop())
+                    continue
+                if paused:
+                    paused = False
+                    self.log.info(
+                        "Rotator data received; resume making "
+                        "the camera cable wrap follow the rotator"
+                    )
+                    await self.send_command(
+                        commands.CameraCableWrapEnableTracking(on=True)
+                    )
                 position, velocity, tai = position_velocity_tai
 
                 command = commands.CameraCableWrapTrack(
                     position=position, velocity=velocity, tai=tai,
                 )
                 await self.send_command(command)
+                self.evt_cameraCableWrapTarget.set_put(
+                    position=position, velocity=velocity, taiTime=tai
+                )
 
         except asyncio.CancelledError:
-            self.log.info("Camera cable wrap control ends")
+            self.log.info("Camera cable wrap following ends")
+        except salobj.ExpectedError as e:
```

Comment:
Don't any of these should take the CSC to `FAULT` state?

Reply:
Not yet. It's a good question.